ci/e2e-tests/az-login.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+
+echo 'Installing and updating azure-iot extension...' >&2
+az extension add --name azure-iot
+az extension update --name azure-iot
+echo 'Installed and updated azure-iot extension' >&2
+
+
+echo 'Logging in to Azure...' >&2
+>/dev/null az login --service-principal \
+    --tenant "$AZURE_TENANT_ID" \
+    --username "$AZURE_USERNAME" \
+    "--password=$AZURE_PASSWORD"
```

Comment:
Any reason that "--password arg" is quoted but tenant and username only have the arg quoted?

Reply:
IIRC there was a reason I had to use `--password=foo` instead of `--password foo`, so the quoting is just a result of that. (This is old code, just moved from the old `e2e-tests.sh`.)

To be precise, bash will allow `--password="$AZURE_PASSWORD"` too, but I find it cleaner to quote entire shell words rather than parts of them.

Reply:
Oh, actually, `az login --help` reminded me:

>```
>    Log in with user name and password. This doesn't work with Microsoft accounts or accounts that
>    have two-factor authentication enabled. Use -p=secret if the first character of the password is
>    '-'.
>        az login -u johndoe@contoso.com -p VerySecret
>```